tools/syz-trace2syz/parser/intermediate_types.go

```diff
@@ -0,0 +1,531 @@
+// E.g. access(....) = -1 ENOENT (No such file or directory). The stuff in the ending parenthesis
+// will get parsed into this type
+type parenthetical struct {
+	tmp string
+}
+
+func newParenthetical() *parenthetical {
+	return &parenthetical{tmp: "tmp"}
+}
+
+// BufferType contains strings
+type BufferType struct {
+	Val string
+}
+
+func newBufferType(val string) *BufferType {
+	return &BufferType{Val: val}
+}
+
+// Name implements IrType Name()
+func (b *BufferType) Name() string {
+	return bufferTypeName
+}
+
+// String implements IrType String()
+func (b *BufferType) String() string {
+	return fmt.Sprintf("Buffer: %s with length: %d\n", b.Val, len(b.Val))
+}
+
+type intType struct {
+	Val int64
+}
+
+func newIntType(val int64) (typ *intType) {
+	return &intType{Val: val}
+}
+func (i *intType) eval(target *prog.Target) uint64 {
+	return uint64(i.Val)
+}
+
+func (i *intType) string() string {
+	return strconv.Itoa(int(i.Val))
+}
+
+// Flags contains set of flagTypes. Most of the time will contain just 1 element
+type Flags []*flagType
+
+// Ints Contains set of intTypes. Most of the time will contain just 1 element
+type Ints []*intType
+
+// NewIntsType - Constructor
+func NewIntsType(vals []int64) Ints {
+	var ints []*intType
+	for _, v := range vals {
+		ints = append(ints, newIntType(v))
+	}
+	return ints
+}
+
+// Eval implements Expression's Eval()
+func (f Flags) Eval(target *prog.Target) uint64 {
+	a := make([]*flagType, len(f))
+	copy(a, f)
+	val := uint64(0)
+	for len(a) > 0 {
+		val |= a[0].eval(target)
+		a = a[1:]
+	}
+	return val
+}
+
+// Name implements IrType Name()
+func (f Flags) Name() string {
+	return exprTypeName
+}
+
+// String implements IrType String()
+func (f Flags) String() string {
+	a := make([]*flagType, len(f))
+	var buf bytes.Buffer
+	copy(a, f)
+	for len(a) > 1 {
+		buf.WriteString(a[0].string())
+		buf.WriteString(" ")
+		a = a[1:]
+	}
+	buf.WriteString(a[0].string())
+	return buf.String()
+}
+
+// Eval implements Expression's Eval()
+func (i Ints) Eval(target *prog.Target) uint64 {
+	if len(i) > 1 {
+		// We need to handle this case by case. We allow more than one elemnt
+		// just to properly parse the traces
+		log.Fatalf("Cannot evaluate Ints with more than one element")
+	}
+	if len(i) == 1 {
+		return i[0].eval(target)
+	}
+	return 0
+}
+
+// Name implements IrType Name()
+func (i Ints) Name() string {
+	return exprTypeName
+}
+
+// String implements IrType String()
+func (i Ints) String() string {
+	if len(i) == 1 {
+		return i[0].string()
+	}
+	return ""
+}
+
+type flagType struct {
+	Val string
+}
+
+func newFlagType(val string) (typ *flagType) {
+	return &flagType{Val: val}
+}
+
+func (f *flagType) eval(target *prog.Target) uint64 {
+	flag := f.string()
+	if trueFlag, ok := config.Consts[flag]; ok {
+		flag = trueFlag
+	}
+	if val, ok := target.ConstMap[flag]; ok {
+		return val
+	}
+	if _, ok := config.Ignore[flag]; ok {
+		return 0
+	}
+	log.Fatalf("Failed to eval flag: %s\n", flag)
```

Comment:
Please remove the remaining `\n` in log calls, this query helps:
```
syz-trace2syz$ find -name "*.go" -exec grep "log\..*\\n\"" {} \; -print
		log.Fatalf("Cannot parse type %#v for Buffer Type\n", traceType)
		log.Fatalf("Resource Type only supports Expression")
		log.Fatalf("Unsupported Type for Proc: %#v\n", traceType)
./proggen/proggen.go
		log.Fatalf("Preprocess modifyldt received unexpected strace type: %s\n", a.Name())
		log.Fatalf("shmctl could not evaluate size of buffer: %#v\n", a)
./proggen/preprocess_hooks.go
		log.Fatalf("Madvise advice arg is not expression")
		log.Fatalf("Parsing Mmap length but type is not expression")
		log.Fatalf("Failed to Parse Fd because type is not Expression")
./proggen/vma_call_handlers.go
	log.Fatalf("Failed to eval flag: %s\n", flag)
./parser/intermediate_types.go
		log.Logf(2, "Failed to decode string: %s, with error: %s\n", s, err.Error())
./parser/lex.go
```